src/accelerate/state.py

```diff
@@ -708,13 +708,12 @@ def __init__(
                     self.distributed_type = DistributedType.MEGATRON_LM
                     megatron_lm_plugin.set_mixed_precision(self._mixed_precision)
                     self.megatron_lm_plugin = megatron_lm_plugin
-            elif self.distributed_type in [DistributedType.MULTI_CPU, DistributedType.NO]:
+            elif self.distributed_type in [DistributedType.MULTI_CPU, DistributedType.MULTI_XPU, DistributedType.NO]:
                 if self.device.type == "cpu" and ipex_plugin is not None:
                     self.ipex_plugin = ipex_plugin
                     if self.ipex_plugin is not None:
                         self.ipex_plugin.set_mixed_precision(mixed_precision)
-            if self.distributed_type in [DistributedType.MULTI_XPU, DistributedType.NO]:
-                if self.device.type == "xpu" and ipex_plugin is not None:
+                elif self.device.type == "xpu" and ipex_plugin is not None and is_xpu_available():
                     self.ipex_plugin = ipex_plugin
                     if self.ipex_plugin is not None:
                         self.ipex_plugin.set_mixed_precision(mixed_precision)
```

Comment:
This could all be simplified further, refactoring all of L711->719 to be:

```python
elif self.distributed_type in [DistributedType.MULTI_CPU, DistributedType.MULTI_XPU, DistributedType.NO]:
    if (
        self.device.type == "cpu" 
        or (self.device.type == "xpu" and is_xpu_available())
        and ipex_plugin is not None
    ):
        self.ipex_plugin = ipex_plugin
        self.ipex_plugin.set_mixed_precision(mixed_precision)
```

Reply:
Yes had thought of doing so 😄